src/renderer/components/+workloads-pods/pod-details-secrets.tsx

```diff
@@ -17,18 +17,27 @@ export interface PodDetailsSecretsProps {
 }
 
 export const PodDetailsSecrets = observer(({ pod }: PodDetailsSecretsProps) => {
-  const [secrets] = useState(observable.map<string, Secret>());
-
-  useEffect(() => autorun(async () => {
-    const podSecrets = await Promise.all(
-      pod.getSecrets().map(secretName => secretsApi.get({
-        name: secretName,
-        namespace: pod.getNs(),
-      })),
-    );
-
-    secrets.replace(podSecrets.filter(Boolean).map(secret => [secret.getName(), secret]));
-  }), []);
+  const [secrets, setSecrets] = useState(new Map<string, Secret>());
+
+  useEffect(() => (
+    reaction(
+      () => pod.getSecrets(),
+      async (secretNames) => {
+        const results = await Promise.allSettled(
+          secretNames.map(secretName => secretsApi.get({
+            name: secretName,
+            namespace: pod.getNs(),
+          })),
+        );
+
+        setSecrets(new Map(
+          results
+            .filter(result => result.status === "fulfilled" && result.value)
+            .map(result => (result as PromiseFulfilledResult<Secret>).value)
+            .map(secret => [secret.getName(), secret]),
+        ));
```

Comment:
Something wrong here? The secret `storage-provisioner-token-tltgv` is not a link (it is in 5.4.4):

![Screen Shot 2022-03-25 at 11 02 59 AM](https://user-images.githubusercontent.com/40840436/160146893-c7dc3058-0ef4-45f4-a4bc-1a54f22f96da.png)

Reply:
Yes, I forgot `fireImmediately: true`